### DISCUSSION TITLE
Instance Uptime - Understanding

### DISCUSSION THREAD

**Discussion by AGlennDa** [trimble-oss/dba-dash#1083] [Q&A] [answered]
I might be stupid, but could you please explain to me, why the instance uptime is mentioned like "15 Mins (+39mins)"
When clicking the information, I am able to see the exact time that the machine has been up, which is the 15 + 30 minutes. 
But why is it written like this? :)

**Comment by DavidWiseman** [comment, marked as answer]
I agree, this can be a little confusing.  The reason is because the SQL Server start time is captured as part of the [OSInfo](https://raw.githubusercontent.com/trimble-oss/dba-dash/refs/heads/main/DBADash/SQL/SQLOSInfo.sql) collection which runs every 1hr.  At the time of the collection, the instance was up for 15min which was 39 minutes ago.  The uptime _should_ currently be 54 minutes but it could also have been restarted sometime in the last 39 minutes that we don't know about yet.  

This confusion would go away by checking the SQL server start time more frequently at the cost of some extra overhead in running the collection more frequently.  The collection is very low overhead so it could be ran more frequently - or maybe just the sqlserver_start_time part of the collection.  

If you wanted you could adjust the schedule of the OSInfo collection to run more frequently - maybe every 1min instead of 1hr.    

I might consider adjusting the default schedule in future.

**Reply by AGlennDa** [reply]
Hi David, 
Thank you for the answer :D 
How about pulling the timestamp of the tempdb-file created on a service restart? 
**select name, create_date from sys.databases where name='tempDB'**

**Reply by DavidWiseman** [reply]
Thanks for the script.  The issue is just collecting this data more frequently rather than how to collect it. On newer versions of SQL Server, the start time is available more directly from sys.dm_os_sys_info.  Older SQL versions (SQL 2005!) you can use the create_date of tempdb like the query you provided.

The [OS Info](https://raw.githubusercontent.com/trimble-oss/dba-dash/refs/heads/main/DBADash/SQL/SQLOSInfo.sql) collection pulls the data from sys.dm_os_sys_info and falls back to create_date of tempdb if needed.  The collection is sub 1ms in my lab so running it more frequently shouldn't be a problem.  I might change the default in a future version or the schedule can be customized manually in the service config tool.
